### ISSUE TITLE
Only load http/https URLs when scanning a QR code from the URL bar

### ISSUE THREAD

**Issue by SergeyZhukovsky** [brave/brave-browser#28703]
We need to do the same as described in iOS issue on Android for QR code reader https://github.com/brave/brave-ios/issues/6997

**Comment by kjozwiak** [comment]
The above requires `1.48.173` or higher for `1.48.x` verification 👍

**Comment by kjozwiak** [comment]
Even though the above was merged into the `1.48.x` branch, moving this issue into the `1.49.x` milestone as we didn't have another `1.48.x` release. It was uplifted into `1.49.x` & `1.48.x` via https://github.com/brave/brave-core/pull/17351 & https://github.com/brave/brave-core/pull/17352. We'll be closing/removing the `1.49.x - Release #6` milestone.

**Comment by kjozwiak** [comment]
The above can be verified with `1.49.120` as this landed a while ago via `1.49.x` but was waiting for a potential `1.48.x` release as mentioned above.

**Comment by Uni-verse** [comment]
In progress on Samsung GS 21 5G using version(s):

```
Brave	1.49.120 Chromium: 111.0.5563.64 (Official Build) (64-bit) 
Revision	c710e93d5b63b7095afe8c2c17df34408078439d-refs/branch-heads/5563@{#995}
OS	Android 13;
```
